docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+<!-- @@author nghochi123 -->
+### Storage
+
+The `Storage` class is responsible for the serialization of `Entry` data into a csv-like syntax, as well as the deserialization of that data back into `Entry` objects.
+
+The main callable functions to be used are:
+
+- `readFromDatabase()` - Deserializes data stored in text form back into `Entry` objects. Executed when PocketPal is instantiated
+- `writeToDatabase()` - Serializes `Entry` objects in `EntryLog` into text form.
+- `reset()` - Clears whatever is in the stored text file, without affecting what is in the current `EntryLog`.
+
+The structure of the Storage class is as follows:
+
+![StorageClassDiagram](static/backend/storage/StorageClassDiagram.png)
+
+The Sequence Diagram below illustrates the interactions within the `Parser` component upon initialization of PocketPal, as well as whenever data is being saved.
+
+![StorageSequenceDiagram](static/backend/storage/StorageSequenceDiagram.png) 
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+<!-- @@author -->
+
+<!-- @@author jinxuan-owyong -->
+### API
+#### Endpoints
+
+![Endpoints](static/backend/endpoint/EndpointClassDiagram.png)
+
+The sequence diagram for specific request handling at each endpoint can be viewed at their respective sections.
+
+Each endpoint is a child class `Endpoint`. Currently, there are 2 endpoints available:
+
+| Endpoint   | Method to call          |
+| ---------- | ----------------------- |
+| `/entry`   | `callEntryEndpoint()`   |
+| `/entries` | `callEntriesEndpoint()` |
+
+##### Creating a request
+
+- To create a request, simply instantiate `pocketpal.communication.pocketpal.Request`
+  with the desired request method.
+- If there are any parameters associated with the request, you may add them using `addParam()`
+
+```java
+Request req = new Request(RequestMethod.PATCH);
+req.addParam(RequestParams.EDIT_DESCRIPTION, "mango juice");
+```
+
+##### Making a request
+
+- To call each endpoint, pass the `Request` into its corresponding method in the frontend.
+- A `Response` will be returned to the frontend.
+
+> All request body and parameter data should be serialised with `String.valueOf()` if not specified.
+
+```java
+Backend backend = new Backend();
+Response res = backend.callEntryEndpoint(req);
+
+if (res.getResponseStatus() != ResponseStatus.OK) {
+   // handle status        
+}
+
+Entry entry = EntryParser.deserialise(res.getData());
+// process entry
+```
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+#### Access all entries available
+##### Get recent or all entries
+
+<details>
+   <summary>Sequence diagram</summary>
+
+   <img alt="Entries Endpoint [GET] Sequence Diagram" src="docs/../static/backend/EntriesEndpointGetSequence.png" />
```

Comment:
The sequence diagram here is broken and cannot be viewed, so are most of the diagrams in this section.